src/engine.rs

```diff
@@ -686,6 +693,18 @@ mod tests {
                 self.file_count(Some(LogQueue::Append)) + self.file_count(Some(LogQueue::Rewrite))
             }
         }
+
+        fn stale_file_count(&self, queue: Option<LogQueue>) -> usize {
```

Comment:
Don't expose the concept of stale. If you must, use `fs.file_count() - pipe.file_count(None)` instead.